gqlalchemy/under-the-hood/python-graph-translators.md

```diff
@@ -0,0 +1,123 @@
+    memgraph.execute(query)
+```
+
+### `get_instance()` method
+
+The `get_instance()` method translates data stored inside Memgraph into NetworkX graph. It traverses the graph and it stores node and relationship objects along with their properties in a NetworkX DiGraph object. Since NetworkX doesn't support node labels and relationship type in a way Memgraph does, they are encoded as node and edge properties, as values of `label` and `type` key. To create NetworkX graph from data stored in Memgraph, following code can be run:
+
+```
+graph =  NxTranslator().get_instance()
+```
+
+## PyG graph translator
+
+The `PyGTranslator` class implements the PyG graph translator and inherits from the `Translator` class. The `PyGTranslator` class can be imported from the `gqlalchemy.transformations.translators.pyg_translator` module. 
+
+[![docs-source](https://img.shields.io/badge/source-PyG%20Translator-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/transformations/translators/pyg_translator.py)
+
+Translating the graph means that you can **import** PyG graph into Memgraph as well as **export** data from Memgraph into PyG graph in your Python code. The `PyGTranslator` defines two important methods:
+ 
+- [`to_cypher_queries()`](#to_cypher_queries-method-1) - The method which generates Cypher queries to create a graph in Memgraph.
+- [`get_instance()`](#get_instance-method-1) - The method which creates PyG instance from the graph stored in Memgraph. 
+
+### `to_cypher_queries()` method
+
+The `to_cypher_queries()` method produces Cypher queries to create graph objects in Memgraph for both homogeneous and heterogeneous graph. This method can translate one-dimensional as well as multidimensional features to Memgraph properties. Isolated nodes in the graph won't get translated into Memgraph. Nodes and relationships will have property `pyg_id` set to the id they have as part of the PyG graph for the consistency reasons. To run the generated queries, following code can be used:
+
+```
+for query in PyGTranslator().to_cypher_queries(pyg_graph):
+    memgraph.execute(query)
+```
+
+
+### `get_instance()` method
+
+The `get_instance()` method returns instance of PyG heterograph from all relationships stored in Memgraph. Isolated nodes are ignored because they don't contribute in message passing neural networks. Only numerical properties that are set on all nodes and relationships are translated to the PyG instance since that is PyG requirement. Hence, any string properties, as well as numerical properties that aren't set on all nodes or relationships, won't be translated to the PyG instance. However, properties of type list will be translated to the PyG instance as a feature. Regardless of how data is connected in Memgraph, the returned PyG graph will be a heterograph instance. To create PyG graph from data stored in Memgraph, following code can be run:
```

Comment:
```suggestion
The `get_instance()` method returns an instance of PyG heterograph from all relationships stored in Memgraph. Isolated nodes are ignored because they don't contribute to message passing neural networks. Only numerical properties that are set on all nodes and relationships are translated to the PyG instance since that is PyG requirement. Hence, any string properties, as well as numerical properties that aren't set on all nodes or relationships, won't be translated to the PyG instance. However, properties of type list will be translated to the PyG instance as a feature. Regardless of how data is connected in Memgraph, the returned PyG graph will be a heterograph instance. To create PyG graph from data stored in Memgraph, the following code can be run:
```